src/main/java/com/product/dao/CustomerDaoImpl.java

```diff
@@ -0,0 +1,34 @@
+package com.product.dao;
+
+import java.net.MalformedURLException;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Repository;
+import org.springframework.web.client.RestTemplate;
+
+import com.product.exception.CustomerProductException;
+import com.product.model.CustomerResponse;
+
+@Repository
+public class CustomerDaoImpl implements CustomerDao {
+
+	private RestTemplate restTemplate;
+
+	@Autowired
+	CustomerDaoImpl(RestTemplate restTemplate) {
+		this.restTemplate = restTemplate;
+	}
+
+	@Override
+	public ResponseEntity<CustomerResponse> getCustomerResponseById(int id) throws MalformedURLException {
+		try {
+			ResponseEntity<CustomerResponse> response = restTemplate
```

Comment:
`CustomerResponse` shouldnt be in ProductService... It should be just `Customer`

Reply:
Done.